contracts/token/ERC20/SafeERC20.sol

```diff
@@ -38,6 +38,27 @@ library SafeERC20 {
   )
     internal
   {
+    require((value == 0) || (token.allowance(msg.sender, spender) == 0));
```

Comment:
We may want to add a comment here, saying something like:

'safeApprove should only be called when setting an initial allowance, or when resetting it to zero. To increase and decrease it, use safeIncreaseAllowance and safeDecreaseAllowance.'

Reply:
will add that.